documentation/modules/ROOT/pages/dev-services.adoc

```diff
@@ -0,0 +1,221 @@
+= Dev Services
+
+Quarkus supports the automatic provisioning of unconfigured services in *development and test* mode. 
+We refer to this capability as Dev Services. From a developer's perspective this means that if you include an extension and don't configure it then Quarkus will automatically start the relevant service (usually using https://www.testcontainers.org/[Testcontainers] behind the scenes) and wire up your application to use this service, even pre-configuring access credentials.
+
+NOTE: Dev Services will only be enabled in dev/test mode, so it will not affect the application running in production.  If you want to disable all Dev Services you can use the quarkus.devservices.enabled=false config property, or you can simply configure the service in which case it will result in the Dev Service being disabled automatically.
+
+TIP: If you are using a proprietary database such as `DB2` or `MSSQL` you will need to accept the license agreement. To do this, create a `src/main/resources/container-license-acceptance.txt` file in your project and add a line with the image name and tag of the database. 
+
+More on zero config setup of datasources can be found https://quarkus.io/guides/datasource#dev-services[here].
+
+== Replace Database Extensions 
+
+Instead of the built-in h2 database, we're now going to use an external database.  Swapping out from one database provider to another is fairly trivial with Quarkus.  We just remove the h2 extension and add the postgresql extension instead:
+
+
+[tabs]
+====
+Maven::
++ 
+--
+[.console-input]
+[source,bash,subs="+macros,+attributes"]
+----
+mvn quarkus:remove-extension -Dextension=quarkus-jdbc-h2
+mvn quarkus:add-extension -Dextension=quarkus-jdbc-postgresql
+----
+
+--
+Quarkus CLI::
++
+--
+[.console-input]
+[source,bash,subs="+macros,+attributes"]
+----
+quarkus extension remove quarkus-jdbc-h2
+quarkus extension add quarkus-jdbc-postgresql
+----
+--
+====
+
+
+== Update configuration
+
+We'll also need to update the configuration and change h2 to postgresql:
+
+[#quarkuspdb-update-props]
+[.console-input]
+[source,config,subs="+macros,+attributes"]
+----
+# Configuration file
+# key = value
+greeting=Hello y'all!
+quarkus.datasource.db-kind=postgresql
```

Comment:
If you only have one jdbc driver in the classpath, it's not necessary to put the property.Maybe I would add this as a TIP

Reply:
good point, I was debating about leaving it in or not.  the tip makes it clear